microceph/ceph/maintenance.go

```diff
@@ -0,0 +1,297 @@
+package ceph
+
+import (
+	"fmt"
+
+	"github.com/canonical/lxd/shared/logger"
+
+	microCli "github.com/canonical/microcluster/v2/client"
+
+	"github.com/canonical/microceph/microceph/client"
+)
+
+// EnterMaintenance put a given node into maintanence mode.
+func EnterMaintenance(clusterClient *microCli.Client, cephClient client.ClientInterface, name string, force, dryRun, setNoout, stopOsds bool) error {
+	ops := []operation{}
+
+	// pre-flight checks
+	if !force {
+		ops = append(ops, []operation{
+			&checkNodeInClusterOps{cephClient, clusterClient},
+			&checkOsdOkToStopOps{cephClient, clusterClient},
+			&checkNonOsdSvcEnoughOps{cephClient, clusterClient, 3, 1, 1},
+		}...)
+	}
```

Comment:
**question:** 

Should we should still run the pre flight-check but ignore the error?

Reply:
good question, how do we want to define "force"? ignore all errors or skip preflight checks?

Reply:
We still run the pre-flight checks but ignore the error.
The ignore error should be logged.